### PR TITLE
[auth] Case sensitive username column

### PR DESCRIPTION
**Pull request by jigold**
Can you point me to where you found `utf8mb4_0900_as_cs` is the correct collation? I'm pretty sure this change doesn't break any foreign key references (I don't think there are any) and just makes it safer for us that a user with a different case name can't impersonate someone else.

### CODE REVIEW

**Review by danking — APPROVED** [review]
This seems right to me

File: auth/sql/change-username-collation.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE users MODIFY COLUMN username varchar(255) NOT NULL COLLATE utf8mb4_0900_as_cs;
+ALTER TABLE users MODIFY COLUMN login_id varchar(255) NOT NULL COLLATE utf8mb4_0900_as_cs;
```
Comment (danking) [inline auth/sql/change-username-collation.sql:2]: you promoted this from `null` to `not null`, I think it will pass if you switch back to `NULL`.

**Review by danking — DISMISSED** [review]

File: auth/sql/change-username-collation.sql (hunk 1 above)
Comment (danking) [reply auth/sql/change-username-collation.sql:2]: it needs a `DEFAULT NULL` actually

**Review by danking — APPROVED** [review]

**Comment by danking** [comment]
Do we have a similar fix for billing project names?

**Comment by jigold** [comment]
#12327

**Pull request merged**